### PR TITLE
roachtest: add smaller index add schemachange, tweak names

### PR DESCRIPTION
**Pull request by dt**
This adds a smaller version of the index add schema change on tpcc test.
It also tweaks the naming as the names became too long and were rejected for TC reasons.
I put the type of schema changes being tested in the name, as this is one of the most identifying aspects of the test.
I removed the node-count, as it is not currently a distinguishing aspect of these tests.
I removed /warehouses= to shorten them to within the length budget.

Release note: none.

### CODE REVIEW

File: pkg/cmd/roachtest/schemachange.go [hunk 1]
```diff
@@ -289,30 +289,66 @@ func findIndexProblem(
...
 			})
 		},
 	})
 }
+
+func registerSchemaChangeIndexTPCC100(r *registry) {
+	warehouses := 100
+	numNodes := 3
+	r.Add(testSpec{
+		Name:    fmt.Sprintf("schemachange/index/tpcc-%d", warehouses),
```
Comment (vivekmenezes) [inline pkg/cmd/roachtest/schemachange.go]: How about adding these two tests in a for loop so we share some common code. The tests can have the tpcc/size suffix and can have separate timeouts.

Comment (dt) [reply pkg/cmd/roachtest/schemachange.go]: Done

**Review by vivekmenezes — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31883)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/roachtest/schemachange.go, line 314 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/31883#-LPkpa2NB2vSxkCQIhBM:-LPkpa2NB2vSxkCQIhBN:bhu9wq) ([raw file](https://github.com/cockroachdb/cockroach/blob/56858b497e300f0ce2aa045652dc0d99638e4723/pkg/cmd/roachtest/schemachange.go#L314)):*
> ```Go
> 						`CREATE INDEX ON tpcc.order (o_carrier_id);`,
> 						`CREATE INDEX ON tpcc.customer (c_last, c_first);`,
> 					})
> ```

The tpcc 1000 test is unlikely to complete with 3 index backfills within the timeout using the current implementation. Perhaps we should run only the first index for now  for 1000 or increase the timeout?

**Review by dt — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31883)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/roachtest/schemachange.go, line 314 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/31883#-LPkpa2NB2vSxkCQIhBM:-LSMIvm42jpWMqT0P7ie:b-7rl0jd) ([raw file](https://github.com/cockroachdb/cockroach/blob/56858b497e300f0ce2aa045652dc0d99638e4723/pkg/cmd/roachtest/schemachange.go#L314)):*
<details><summary><i>Previously, vivekmenezes wrote…</i></summary><blockquote>

The tpcc 1000 test is unlikely to complete with 3 index backfills within the timeout using the current implementation. Perhaps we should run only the first index for now  for 1000 or increase the timeout?
</blockquote></details>

done.

**Review by vivekmenezes — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31883)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Pull request merged**